contracts/test/v0.8/foundry/ocr2dr/OCR2DROracle.t.sol

```diff
@@ -0,0 +1,76 @@
+pragma solidity ^0.8.0;
+
+import {BaseTest} from "../BaseTest.t.sol";
+import {OCR2DROracle} from "../../../../src/v0.8/dev/ocr2dr/OCR2DROracle.sol";
+
+contract OCR2DROracleSetup is BaseTest {
+    OCR2DROracle s_oracle;
+
+    function setUp() public virtual override {
+        BaseTest.setUp();
+
+        s_oracle = new OCR2DROracle();
+    }
+}
+
+contract OCR2DROracle_typeAndVersion is OCR2DROracleSetup {
+    function testTypeAndVersionSuccess() public {
+       assertEq(s_oracle.typeAndVersion(), "OCR2DROracle 0.0.0");
+    }
+}
+
+contract OCR2DROracle_setDONPublicKey is OCR2DROracleSetup {
+    function testSetDONPublicKeySuccess() public {
```

Comment:
Can we have a replica of this function but `_gas`, where all of these assertions, getters and encodings are done outside of that function?